### PR TITLE
[Fortran/gfortran] Enable previously disabled tests

### PR DESCRIPTION
**Pull request by tarunprabhu**
Several tests that previous failed now pass because of more features having been implemented and several bugs having been fixed.

### CODE REVIEW

**Review by clementval — APPROVED** [review]
Thanks!

**Comment by tarunprabhu** [comment]
Could someone test this on AArch64 please. I have tested it on an older machine, but it would be good to test it one something newer before I merge it in.

**Review by luporl — COMMENTED** [review]
I've run the tests on a Ampere Altra Mt Jade and only 2 of them failed.
gdb just says they got a SIGSEGV on Fortran's main.

**Comment by jeanPerier** [comment]
> I've run the tests on an Ampere Altra Mt Jade and only 2 of them failed.
> gdb just says they got a SIGSEGV on Fortran's main.

@tarunprabhu , these are `dg-do run` `dg-shouldfail` tests, you probably do not want to re-enable any such tests which most likely have undefined behaviors for runtime user errors we are not catching. There may be more than the two that @luporl found given the undefined behavior may just randomly segfault or not.

File: Fortran/gfortran/regression/gomp/DisabledFiles.cmake [hunk 1]
```diff
@@ -45,84 +35,49 @@ file(GLOB UNIMPLEMENTED_FILES CONFIGURE_DEPENDS
 
   # unimplemented: derived type components with non default lower bounds
   depend-iterator-1.f90
-  depend-iterator-2.f90
-
-  # unimplemented: assumed rank in procedure interface
-  pr94672.f90
```
Comment (kiranchandramohan) [inline Fortran/gfortran/regression/gomp/DisabledFiles.cmake:51]: Not sure how this started working.

**Comment by kiranchandramohan** [comment]
I see the same results as @luporl on AArch64.
Failed Tests (2):
  test-suite :: Fortran/gfortran/regression/gfortran-regression-execute-regression__pointer_check_11_f90.test
  test-suite :: Fortran/gfortran/regression/gfortran-regression-execute-regression__pointer_check_12_f90.test

**Comment by tblah** [comment]
I also see pointer_check_11.f90 and pointer_check_12.f90 fail on aarch64

**Comment by tarunprabhu** [comment]
Thanks @luporl, @kiranchandramohan, @jeanPerier and @tblah for looking at this. I wrote a script to identify the tests that were passing and automatically enable them. Both that script and the gfortran test suite's build system needs more work as identified by @jeanPerier separately as well. I'll make a manual pass over it and remove any questionable tests in addition to those that you have identified. 

I'll work on the build system after this.

File: Fortran/gfortran/regression/gomp/DisabledFiles.cmake (hunk 1 above)
Comment (tarunprabhu) [reply Fortran/gfortran/regression/gomp/DisabledFiles.cmake:51]: This is expected to raise a compile-time error with the message 

> 'array' not specified in enclosing 'parallel'

In flang, this now errors with the message:

> The DEFAULT(NONE) clause requires that 'array' must be listed in a data-sharing attribute clause

Are these not equivalent? 

It was probably hitting a TODO earlier with the assumed rank error message.

**Comment by tarunprabhu** [comment]
Abandoning this pull request. There are several issues with the test suite as is and I will fix those before enabling more tests.

**Pull request closed without merging**